### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by 3Nakajugo**
create an incident, get all incidents, get a single incident, patch comment and location, delete an incident, create a user,

### CODE REVIEW

File: project/controllers/incident_cntr.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        """
+        new_incident = incident.to_json()
+        incidents.append(new_incident)
+        return incidents
+
+    def get_all_redflags(self):
+        """ 
+        gets all redflags
+        """
+        if len(incidents) > 0:
```
Comment (solnsubuga) [inline project/controllers/incident_cntr.py]: @3Nakajugo where have you declared the `incidents` variable you are checking?

File: project/controllers/incident_cntr.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        for incident in incidents:
+            if incident['incident_id'] == incident_id:
+                return incident
+
+    def delete_single_redflag(self, incident_id):
+        """ 
+        deletes single  redflags
+        """
+        for incident in incidents:
```
Comment (solnsubuga) [inline project/controllers/incident_cntr.py]: @3Nakajugo Where have you declared the incidents variable you are checking here?

Comment (3Nakajugo) [reply project/controllers/incident_cntr.py]: incidents is a list I imported from the modal

File: project/controllers/user_cntr.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+class UserController:
+
+    def create_user(self, user):
+        new_user = user.user_to_json()
+        users.append(new_user)
+        return users
+
+    def get_all_users(self):
+        for user in users:
```
Comment (solnsubuga) [inline project/controllers/user_cntr.py:12]: In this function just return the entire users list
```suggestion
        return users
```

File: project/models.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            "file": self.file,
+            "comment": self.comment
+        }
+
+
+class User:
+    """
+    class for users
+    """
+    def __init__(self, *args):
```
Comment (solnsubuga) [inline project/models.py:41]: use keyword arguments here
```suggestion
    def __init__(self, **kwargs):
```

File: project/models.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        self.user_id = len(users)+1
+        self.first_name = args[0]
```
Comment (solnsubuga) [inline project/models.py:43]: If you are using kwargs this also changes
```suggestion
        self.first_name = kwargs.get('first_name')
```

File: tests/test_app.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+                         "no incident with such an id")
+    
+    def test_posting_missing_field(self):
+        response = self.test_client.post(
+            '/api/v1/incidents', data=json.dumps(self.incident_missing))
+        self.assertEqual(response.status_code, 400)
+
+
+
+    # def test_edit_location_of_redlag_that_doesnot_exist(self):
```
Comment (solnsubuga) [inline tests/test_app.py:109]: @3Nakajugo remove this commented out block of code.

**Pull request closed without merging**